cristin-organization/src/main/java/no/unit/nva/cristin/organization/CristinApiClient.java

```diff
@@ -45,15 +52,65 @@ public Organization getOrganization(URI uri)
      *
      * @param requestQueryParams Map containing verified query parameters
      */
-    public SearchResponse<Organization> queryInstitutions(Map<String, String> requestQueryParams) {
-        return new SearchResponse<Organization>(
-                new UriWrapper(HTTPS, DOMAIN_NAME)
-                        .addChild(BASE_PATH)
-                        .addChild(UriUtils.INSTITUTION)
-                        .addQueryParameters(requestQueryParams)
-                        .getUri())
-                .withProcessingTime(0L)
-                .withSize(0)
-                .withHits(Collections.emptyList());
+    public SearchResponse<Organization> queryOrganizations(Map<String, String> requestQueryParams)
+            throws NotFoundException, FailedHttpRequestException {
+        Map<String, String> cristinRequestQueryParams = translateToCristinApi(requestQueryParams);
+        URI queryUri = new UriWrapper(HTTPS, CRISTIN_API_BASE)
+                .addChild(UNITS_PATH)
+                .addQueryParameters(cristinRequestQueryParams)
+                .getUri();
+
+        long start = System.currentTimeMillis();
+        SearchResponse<Organization> searchResponse = httpExecutor.query(queryUri);
+
+        return updateSearchResponseMetadata(searchResponse,requestQueryParams, System.currentTimeMillis() - start);
+    }
+
+    private Map<String, String> translateToCristinApi(Map<String, String> requestQueryParams) {
+        return Map.of(
+                CRISTIN_QUERY_NAME_PARAM, requestQueryParams.get(QUERY),
+                PAGE, requestQueryParams.get(PAGE),
+                CRISTIN_PER_PAGE_PARAM, requestQueryParams.get(NUMBER_OF_RESULTS));
     }
+
+    private SearchResponse<Organization> updateSearchResponseMetadata(
+            SearchResponse<Organization> searchResponse,
+            Map<String, String> requestQueryParams,
+            long timeUsed) {
+        final URI baseUri = new UriWrapper(HTTPS, DOMAIN_NAME).addChild(BASE_PATH).addChild(ORGANIZATION_PATH).getUri();
+        searchResponse.setId(new UriWrapper(baseUri).addQueryParameters(requestQueryParams).getUri());
```

Comment:
refactor this command to a method `new UriWrapper(baseUri).addQueryParameters(requestQueryParams).getUri()`